lib/src/main/java/com/thenewboston/api/common/PostDataSource.kt

```diff
@@ -10,14 +10,34 @@ import com.thenewboston.data.dto.bankapi.crawl.response.Crawl
 import com.thenewboston.data.dto.bankapi.upgradenoticedto.UpgradeNoticeRequest
 import com.thenewboston.data.dto.bankapi.validatorconfirmationservicesdto.ConfirmationServices
 import com.thenewboston.data.dto.bankapi.validatorconfirmationservicesdto.request.PostConfirmationServicesRequest
+import com.thenewboston.data.dto.primaryvalidatorapi.bankblockdto.BankBlock
+import com.thenewboston.data.dto.primaryvalidatorapi.bankblockdto.request.BankBlockRequest
 import com.thenewboston.utils.BankAPIEndpoints
+import com.thenewboston.utils.PrimaryValidatorAPIEndpoints
 import io.ktor.client.request.*
 import io.ktor.client.statement.*
 import io.ktor.utils.io.errors.*
 import javax.inject.Inject
 
 class PostDataSource @Inject constructor(private val networkClient: NetworkClient) {
 
+    suspend fun doSendBankBlock(request: BankBlockRequest): Outcome<BankBlock> {
+        val response = networkClient.defaultClient.post<BankBlock> {
+            url(PrimaryValidatorAPIEndpoints.BANK_BLOCKS_ENDPOINT)
+            body = request
+        }
+
+        return when {
+            response.accountNumber.isBlank() -> {
+                val nodeIdentifier = request.nodeIdentifier
+                val message =
+                    "Received invalid response sending bank block with node identifier: $nodeIdentifier"
+                return Outcome.Error(message, IOException())
+            }
+            else -> Outcome.Success(response)
```

Comment:
No need to check account number is blank. The NetworkUtils class will handle the error scenario where if the call to the endpoint fails, it will throw the Error outcome. 

I know some of the other functions do this, but we will be removing these cases in the future as they are unnecessary since the API should never return empty account numbers.